src/test/ui/macros/assert-format-lazy.rs

```diff
@@ -0,0 +1,11 @@
+// run-pass
+
+#[allow(unreachable_code)]
+fn main() {
+    assert!(true, "Failed: {:?}", panic!("assert! evaluated format expressions"));
+    debug_assert!(true, "Failed: {:?}", panic!("debug_assert! evaluated format expressions"));
```

Comment:
Are debug assertions disabled for tests? If so, this doesn't test what is intended.

Reply:
I believe it does right? https://github.com/rust-lang/rust/blob/0d97f7a96877a96015d70ece41ad08bb7af12377/src/tools/clippy/tests/ui/debug_assert_with_mut_call.rs#L31

Reply:
From trying it out, it looks like the debug assertions were indeed disabled for this test, so I updated it to ensure they're enabled. (I verified that the test now fails when `debug_assert!(true, ...)` is changed to `debug_assert!(false, ...)`.